File: changelog_builder/commit_processor/commit_length.py
class CommitLength:
    def __init__(self, config):
        self.config = config

    def process(self, commits):
        if 'full' == self.config['commit_length']:
            return commits

        short_commits = {}
        for key in commits:
            commit = commits[key]
            lines = commit.split("\n")

            if len(lines[0]) > 2:
                short_commits[key] = lines[0]
            else:
                short_commits[key] = lines[1]

        return short_commits

File: lib/processor/commitlength.js
var config = require('../../config');

function CommitLength() {

}

CommitLength.prototype.process = function(commits) {
	if ('full' == config.commit_length) return commits;

	var shortCommits = {};
	for (var key in commits) {
		if (commits.hasOwnProperty(key)) {
			var commit = commits[key];
			var lines = commit.split("\n");

			if (lines[0].length > 2) {
				shortCommits[key] = lines[0];
			} else {
				shortCommits[key] = lines[1];
			}
		}
	}

	return shortCommits;
}

module.exports = CommitLength;
